src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -346,6 +350,12 @@ public async Task<IActionResult> Create(EditTenantViewModel model)
                 shellSettings["Secret"] = Guid.NewGuid().ToString();
                 shellSettings["RecipeName"] = model.RecipeName;
                 shellSettings["FeatureProfile"] = model.FeatureProfile;
+                // To distinguish between null and empty string, never store null in the shellSettings for TablePrefixSeparator
+                // otherwise old tenants will have invalid table names.
+                shellSettings["TablePrefixSeparator"] = model.TablePrefixSeparator ?? String.Empty;
```

Comment:
You sure about that? I tested it and was able to store empty string. If it is null, then it won't be part of the output. Please double check

Reply:
Yes I already checked it many times, if then you setup the tenant you will see that it still uses the '_' separator, not an empty one. Also take care about one of my other comments about the schema that should be passed through e.g. `UseSqlServer()`, otherwise changing it has no effect.

> If it is null, then it won't be part of the output.

It is and then rendered as an empty string